src/components/accordion/accordion.pw.tsx

```diff
@@ -0,0 +1,735 @@
+            messages.push(data);
+          }}
+        />
+      );
+
+      await accordionTitleContainer(page).click();
+
+      await expectEventWasCalledOnce(messages, "onClick");
+    });
+  });
+
+  testData.forEach((titleValue) => {
+    test(`should render Accordion component with ${titleValue} as a title`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<AccordionComponent title={titleValue} />);
+
+      await expect(accordionTitleContainer(page)).toContainText(titleValue);
+    });
+  });
+
+  testData.forEach((titleValue) => {
+    test(`should render Accordion component with ${titleValue} as a subtitle`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<AccordionComponent subTitle={titleValue} />);
+
+      await expect(accordionTitleContainer(page)).toContainText(titleValue);
+    });
+  });
+
+  ([
+    [SIZE.SMALL, 24],
+    [SIZE.LARGE, 45],
+  ] as const).forEach(([size, height]) => {
+    test(`should render Accordion component with ${size} as a size and has height property set to ${height}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<AccordionComponent size={size} />);
+
+      const cssHeight = await getStyle(accordionTitleContainer(page), "height");
+      await expect(parseInt(cssHeight)).toBeLessThanOrEqual(height + 1);
+      await expect(parseInt(cssHeight)).toBeGreaterThanOrEqual(height - 1);
+    });
+  });
+
+  ([
+    ["full", "solid", "rgb(204, 214, 219)"],
+    ["none", "none", "rgba(0, 0, 0, 0.9)"],
+  ] as const).forEach(([borderProp, borderStyle, colour]) => {
+    test(`should render Accordion component with border type '${borderProp}'`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<AccordionComponent borders={borderProp} />);
+
+      await expect(accordion(page)).toHaveCSS(
+        "border-bottom-style",
+        borderStyle
+      );
+      await expect(accordion(page)).toHaveCSS("border-bottom-color", colour);
+    });
+  });
+
+  [true, false].forEach((state) => {
+    test(`should render Accordion component with default expanded state '${state}'`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<AccordionComponent defaultExpanded={state} />);
+
+      await expect(accordionTitleContainer(page)).toHaveAttribute(
+        "aria-expanded",
+        String(state)
+      );
+      await expect(accordionTitleContainer(page)).toBeVisible();
+    });
+  });
+
+  [true, false].forEach((state) => {
+    test(`should render Accordion component with expanded state '${state}'`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<AccordionComponent expanded={state} />);
+
+      await expect(accordionTitleContainer(page)).toHaveAttribute(
+        "aria-expanded",
+        String(state)
+      );
+      await expect(accordionTitleContainer(page)).toBeVisible();
+    });
+  });
+
+  ([
+    ["white", "rgb(255, 255, 255)", "be.visible"],
+    ["transparent", "rgba(0, 0, 0, 0)", "be.visible"],
```

Comment:
Should third parameter "be.visible" be here?

Reply:
good spot, thanks. Looks like I forgot to remove this from the Cypress test even though we're not using it.